fastcrud/endpoint/endpoint_creator.py

```diff
@@ -221,26 +222,45 @@ def __init__(
         }
         self.endpoint_names = {**self.default_endpoint_names, **(endpoint_names or {})}
 
+    def _base_statement(self):
+        """Generic statement used across router endpoints unless overridden
+        in specific method.
+        """
+        return
+
+    def _base_read_statement(self):
+        """Create sql statement for object reading."""
+        return self._base_statement()
+
+    def _base_create_statement(self):
+        """Create sql statement for object creation."""
+        return self._base_statement()
+
+    @staticmethod
+    def _base_filter(stmt):
+        """Generic filter used across router endpoints unless overridden
+        in specific method.
+        """
+        return stmt
+
+    def _read_filter(self, stmt):
+        """Read specific filter."""
+        return self._base_filter(stmt)
+
     def _create_item(self):
         """Creates an endpoint for creating items in the database."""
 
         async def endpoint(
             db: AsyncSession = Depends(self.session),
             item: self.create_schema = Body(...),  # type: ignore
         ):
-            unique_columns = _extract_unique_columns(self.model)
```

Comment:
I'm not a fan of this exists check - it generates a number of database calls. In fact, just tested that every "unique" field in SQLAlchemy introduces an index on that field, so the responses are immediate. Anyway, I still find it sufficient to just handle generic IntegrityError exception as presented.